src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -58,10 +58,8 @@ class LinalgBackendBase
 	 * as \f$\sum_i a_i b_i\f$
 	 */
 	#define BACKEND_GENERIC_DOT(Type) \
-	virtual Type dot(const SGVector<Type>&, const SGVector<Type>&) const \
-	{  \
-		SG_SNOTIMPLEMENTED; \
-	}
+	virtual Type dot(const SGVector<Type>&, const SGVector<Type>&) const = 0;\
```

Comment:
Nono, wait.
The linalg operations should not be purely virtual (we allow for incomplete backend implementations)
Only the to_gpu should be purely virtual as this is really needed when implementing a GPU backend